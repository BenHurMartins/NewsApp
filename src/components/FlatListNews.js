import React, {useEffect, useState} from 'react';
import {FlatList, StyleSheet, Text} from 'react-native';
import {Mixins} from '../styles';
import {getNews} from '../services/news/news';
import {useSelector, useDispatch} from 'react-redux';
import {NewsItemComponent} from './';
import {ADD_PAGE} from '../reducers/actionsTypes';

const FlatListNews = () => {
  const dispatch = useDispatch();
  const {sourceId, page} = useSelector(state => state.NewsReducer);
  const [news, setNews] = useState([]);
  const renderItem = ({item}) => (
    <NewsItemComponent key={item.url} news={item} />
  );

  useEffect(async () => {
    console.log('sourceId', sourceId);
    await getNews(sourceId, page).then(response => {
      console.log('News', response.data.articles.length);
      setNews(response.data.articles);
    });
  }, [sourceId]);

  const getMoreNews = async () => {
    await getNews(sourceId, page + 1).then(response => {
      dispatch({type: ADD_PAGE});
      console.log('Chamou aqui');
      const newData = response.data.articles;
      console.log(newData.length);
      setNews(oldData => [...oldData, ...newData]);
    });
  };

  return (
    <FlatList
      contentContainerStyle={styles.flatListContainer}
      data={news}
      renderItem={renderItem}
      keyExtractor={(item, index) => item.id}
      onEndReached={getMoreNews}
      onEndReachedThreshold={0.7}
    />
  );
};

export default FlatListNews;

const styles = StyleSheet.create({
  flatListContainer: {
    width: Mixins.DEVICE_WIDTH,
    alignItems: 'center',
  },
});
