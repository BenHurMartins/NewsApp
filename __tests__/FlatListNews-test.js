import 'react-native';
import React from 'react';
import FlatListNews from '../src/components/FlatListNews';
import renderer from 'react-test-renderer';

describe('FlatListNews component', () => {
  it('FlatListNews renders correctly', () => {
    renderer.create(<FlatListNews />);
  });
  it('FlatListNews snapshot test', () => {
    const tree = renderer.create(<FlatListNews />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
