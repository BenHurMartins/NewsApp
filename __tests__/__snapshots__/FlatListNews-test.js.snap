// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatListNews component FlatListNews snapshot test 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "width": 750,
    }
  }
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.7}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
